pkg/authn/oidc.go

```diff
@@ -38,9 +38,13 @@ var (
 
 // NewOIDCAuthenticator returns OIDC authenticator
 func NewOIDCAuthenticator(ctx context.Context, config *OIDCConfig) (*OIDCAuthenticator, error) {
-	dyCA, err := dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
-	if err != nil {
-		return nil, err
+	var dynamicCA *dynamiccertificates.DynamicFileCAContent
+	if len(config.CAFile) > 0 {
+		dyCA, err := dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
```

Comment:
nit:
```suggestion
	if len(config.CAFile) > 0 { // if unset, the OIDC authenticator defaults to host's trust store
	    var err error
		dynamicCA, err = dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
```

Reply:
Addressed with https://github.com/brancz/kube-rbac-proxy/pull/342/commits/84cafebed9010c55eef5950ce21a4cffe2dfe9c6

Should this be squashed into the previous commit?

Reply:
Yes please